nix/status-go/shared/build.nix

```diff
@@ -0,0 +1,141 @@
+{ stdenv, lib, fetchFromGitHub, buildGo114Package
+# Dependencies
+, xcodeWrapper
+, go, androidPkgs
+# metadata and status-go source
+, meta, source
+# build parameters
+, platform ? "android"
+, arch ? "386"
+, api ? "23" }:
+
+let
+  inherit (lib) attrNames getAttr strings concatStringsSep concatMapStrings;
+
+  removeReferences = [ go ];
+  removeExpr = refs: ''remove-references-to ${concatMapStrings (ref: " -t ${ref}") refs}'';
+
+  targetArchMap = rec {
+    "386" = "i686";
+    "arm" = "armv7a";
+    "arm64" = "aarch64";
+  };
+  ldArchMap = {
+    "386" = "x86";
+    "arm" = "arm";
+    "arm64" = "arm64";
+  };
+
+  # Shorthands for the built phase
+  targetArch = getAttr arch targetArchMap;
+  ldArch = getAttr arch ldArchMap;
+
+  osId = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 2;
+  osArch = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 0;
+
+  androidTarget = targetArch + "-linux-" + platform;
+
+  iosArchMap = {
+    "386" = "x86_64";
+    "arm" = "armv7";
+    "arm64" = "arm64";
+  };
+  iosSdk = if arch == "386" then "iphonesimulator" else "iphoneos";
+  iosArch = getAttr arch iosArchMap;
+  compilerFlags = if platform == "android" || platform == "androideabi" then
```

Comment:
You can use [`hasPrefix`](https://github.com/status-im/nixpkgs/blob/3355743cfd8d89cfa8d3b8e6d40c48e62e72d36a/lib/strings.nix#L182-L196) to do this:
```nix
lib.hasPrefix "android" platform
```